tool/metrics.py

```diff
@@ -0,0 +1,113 @@
+"""
+Implements the Distinct-n metric for evaluating diversity in dialogues.
+The implementation is copied from Joko et al. (2024) which is based on Tevet et al. (2021).
+For more details, refer to Joko et al. (2024).
+
+References:
+- Joko et al. (2024): https://dl.acm.org/doi/10.1145/3626772.3657815
+    (GitHub: https://github.com/informagi/laps)
+- Tevet et al. (2021): https://aclanthology.org/2021.eacl-main.25/
```

Comment:
Nit: remove empty first line.

```suggestion
"""Implements the Distinct-n metric for evaluating diversity in dialogues.
The implementation is copied from Joko et al. (2024) which is based on Tevet et al. (2021).
For more details, refer to Joko et al. (2024).

References:
- Joko et al. (2024): https://dl.acm.org/doi/10.1145/3626772.3657815
    (GitHub: https://github.com/informagi/laps)
- Tevet et al. (2021): https://aclanthology.org/2021.eacl-main.25/
```